convert/convert.go

```diff
@@ -0,0 +1,118 @@
+package convert
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kong/deck/file"
+	"github.com/kong/deck/utils"
+	"github.com/kong/go-kong/kong"
+)
+
+type Format string
+
+const (
+	FormatKongGateway Format = "kong-gateway"
+	FormatKonnect     Format = "konnect"
+)
+
+var (
+	AllFormats = []Format{FormatKongGateway, FormatKonnect}
+
+	conversions = map[Format]map[Format]bool{
+		FormatKongGateway: {
+			FormatKonnect: true,
+		},
+	}
+)
+
+func validConversion(from, to Format) (bool, error) {
+	if fromMap, ok := conversions[from]; ok {
+		if _, ok := fromMap[to]; ok {
+			return ok, nil
+		}
+	}
+
+	return false, fmt.Errorf("cannot convert from '%s' to '%s' format", from, to)
+}
+
+func ParseFormat(key string) (Format, error) {
+	format := Format(strings.ToLower(key))
+	switch format {
+	case FormatKongGateway:
+		return FormatKongGateway, nil
+	case FormatKonnect:
+		return FormatKonnect, nil
+	default:
+		return "", fmt.Errorf("invalid format: '%v'", key)
+	}
+}
+
+func Convert(inputFilename, outputFilename string, from, to Format) error {
+	if valid, err := validConversion(from, to); !valid {
+		return err
+	}
+
+	key := from + to
+	switch key {
+	case FormatKongGateway + FormatKonnect:
+		return convertKongGatewayToKonnect(inputFilename, outputFilename)
+	default:
+		return fmt.Errorf("cannot convert from '%s' to '%s' format", from, to)
+	}
+}
+
+func convertKongGatewayToKonnect(inputFilename, outputFilename string) error {
+	inputContent, err := file.GetContentFromFiles([]string{inputFilename})
+	if err != nil {
+		return err
+	}
+
+	for _, service := range inputContent.Services {
+		servicePackage, err := kongServiceToKonnectServicePackage(service)
+		if err != nil {
+			return err
+		}
+		inputContent.ServicePackages = append(inputContent.ServicePackages, servicePackage)
+	}
```

Comment:
(blocker)
I think that we should decouple logic from presentation here.

requested action: Refactor this `for` and the rest of the conversion logic below into a `func` that takes a gateway `file.Content` as an argument and returns a konnect `file.Content` without changing the argument.

If this requires generating a deepcopy func for `type Content`, then it's probably the right time to do that.

Reply:
SGTM. We will need to get https://github.com/Kong/deck/pull/332 merged in first.